src/com/owncloud/android/db/UploadDbHandler.java

```diff
@@ -0,0 +1,416 @@
+
+    static private final String TAG = "UploadDbHandler";
+    static private final String TABLE_UPLOAD = "list_of_uploads";
+
+    // for testing only
+    public void recreateDb() {
+//        getDB().beginTransaction();
+//        try {
+//            mHelper.onUpgrade(getDB(), 0, mDatabaseVersion);
+//            getDB().setTransactionSuccessful();
+//        } finally {
+//            getDB().endTransaction();
+//        }
+    }
+
+    public enum UploadStatus {
+        /**
+         * Upload scheduled.
+         */
+        UPLOAD_LATER(0),
+        /**
+         * Last upload failed. Will retry.
+         */
+        UPLOAD_FAILED_RETRY(1),
+        /**
+         * Upload currently in progress.
+         */
+        UPLOAD_IN_PROGRESS(2),
+        /**
+         * Upload paused. Has to be manually resumed by user.
+         */
+        UPLOAD_PAUSED(3),
+        /**
+         * Upload was successful.
+         */
+        UPLOAD_SUCCEEDED(4),
+        /**
+         * Upload failed with some severe reason. Do not retry.
+         */
+        UPLOAD_FAILED_GIVE_UP(5),
+        /**
+         * User has cancelled upload. Do not retry.
+         */
+        UPLOAD_CANCELLED(6);
+        private final int value;
+
+        private UploadStatus(int value) {
+            this.value = value;
+        }
+
+        public int getValue() {
+            return value;
+        }
+    };
+
+    private UploadDbHandler(Context context) {
+        mDatabaseName = MainApp.getDBName();
+        mHelper = new OpenerHelper(context);
+    }
+
+    private static UploadDbHandler me = null;
+
+    static public UploadDbHandler getInstance(Context context) {
+        if (me == null) {
+            me = new UploadDbHandler(context);
+        }
+        return me;
+    }
+
+    public void close() {
+        getDB().close();
+        setDB(null);
+        me = null;
+    }
+
+    /**
+     * Store a file persistently (to be uploaded later).
+     * 
+     * @param filepath local file path to file
+     * @param account account for uploading
+     * @param message optional message. can be null.
+     * @return false if an error occurred, else true.
+     */
+    public boolean storeFile(String filepath, String account, String message) {
+        // /OBSOLETE
+        Log_OC.i(TAG, "obsolete method called");
+        return false;
+        // ContentValues cv = new ContentValues();
+        // cv.put("path", filepath);
+        // cv.put("account", account);
+        // cv.put("attempt",
+        // UploadStatus.UPLOAD_STATUS_UPLOAD_LATER.getValue());
+        // cv.put("message", message);
+        // long result = mDB.insert(TABLE_UPLOAD, null, cv);
+        // Log_OC.d(TABLE_UPLOAD, "putFileForLater returns with: " + result +
+        // " for file: " + filepath);
+        // return result != -1;
+    }
+
+    // ununsed until now. uncomment if needed.
+    // public Cursor getFailedFiles() {
+    // return mDB.query(TABLE_INSTANT_UPLOAD, null, "attempt>" +
```

Comment:
These are for the old table with only instant uploads, right? Maybe we should just remove it in the upgrade process; something to think about.